src/common/snippets/src/lowered/pass/assign_registers.cpp

```diff
@@ -162,10 +165,12 @@ bool AssignRegisters::run(LinearIR& linear_ir) {
         }
     }
 
-    auto linescan_assign_registers = [](const decltype(live_intervals_vec)& live_intervals,
-                                        const std::set<Reg>& reg_pool) {
-        // http://web.cs.ucla.edu/~palsberg/course/cs132/linearscan.pdf
-        std::map<LiveInterval, Reg, by_ending> active;
+    auto assign_registers = [](const decltype(live_intervals_vec)& live_intervals, const std::set<Reg>& reg_pool) {
+        OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::AssignRegisters::LinearScanOptimized")
+        // Optimized O(n log n) register assignment using priority queue
+        // Priority queue to track active intervals by end time
+        std::priority_queue<std::pair<double, Reg>, std::vector<std::pair<double, Reg>>, std::greater<>> active;
```

Comment:
```suggestion
        using QueueElem = std::pair<decltype(Expression().get_exec_num()), Reg>;
        std::priority_queue<QueueElem, std::vector<QueueElem>, std::greater<>> active;
```

Or you can suggest more informative naming instead of `QueueElem` :)

Reply:
Name is ok for me. Applied